crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -863,7 +878,93 @@ where
             Some(recpts) => recpts,
             None => return Err(EthApiError::UnknownBlockNumber),
         };
-        build_transaction_receipt_with_block_receipts(tx, meta, receipt, &all_receipts)
+
+        #[cfg(feature = "optimism")]
+        let (l1_block_info, l1_fee, l1_data_gas) = {
+            let block = self
+                .provider()
+                .block_by_number(meta.block_number)?
+                .ok_or(EthApiError::UnknownBlockNumber)?;
+
+            let block_timestamp = block.timestamp;
+
+            let l1_block_info: Option<L1BlockInfo> = reth_revm::optimism::parse_l1_info_tx(
+                &block.body.first().ok_or(EthApiError::InternalEthError)?.input()[4..],
+            )
+            .map_err(|_| EthApiError::InternalEthError)
+            .ok();
+            let mut buf = BytesMut::default();
+            tx.encode_enveloped(&mut buf);
+            let data = &buf.freeze().into();
+            if let Some(l1_block_info) = l1_block_info {
+                let l1_fee = (!tx.is_deposit())
+                    .then(|| {
+                        l1_block_info.l1_tx_data_fee(
+                            self.inner.provider.chain_spec(),
+                            block_timestamp,
+                            data,
+                            tx.is_deposit(),
+                        )
+                    })
+                    .transpose()
+                    .map_err(|_| EthApiError::InternalEthError)?;
+                let l1_data_gas = (!tx.is_deposit())
+                    .then(|| {
+                        l1_block_info.l1_data_gas(
+                            self.inner.provider.chain_spec(),
+                            block_timestamp,
+                            data,
+                        )
+                    })
+                    .transpose()
+                    .map_err(|_| EthApiError::InternalEthError)?;
+
+                (Some(l1_block_info), l1_fee, l1_data_gas)
+            } else {
+                (None, None, None)
+            }
+        };
+
+        build_transaction_receipt_with_block_receipts(
+            tx,
+            meta,
+            receipt,
+            &all_receipts,
+            #[cfg(feature = "optimism")]
+            &l1_block_info,
+            #[cfg(feature = "optimism")]
+            l1_fee,
+            #[cfg(feature = "optimism")]
+            l1_data_gas,
+        )
+    }
+
+    /// Helper function for `eth_send_raw_transaction` for Optimism.
```

Comment:
```suggestion
    /// Helper function for `eth_sendRawTransaction` for Optimism.
```

Reply:
Addressed in https://github.com/anton-rs/op-reth/pull/149